src/town_cmd.cpp

```diff
@@ -1087,6 +1087,54 @@ static bool GrowTownWithRoad(const Town *t, TileIndex tile, RoadBits rcmd)
 	return false;
 }
 
+/**
+ * Checks if a town road can be continued on the other side of a bridge.
+ *
+ * @param end_tile The end tile of the bridge
+ * @param bridge_dir The direction of the bridge
+ * @return true if the road can be continued, else false
+ */
+static bool CanRoadContinueAfterBridge(const Town* t, const TileIndex end_tile, const DiagDirection bridge_dir)
+{
+	const int delta = TileOffsByDiagDir(bridge_dir); // +1 tile in the direction of the bridge
+	TileIndex next_tile = end_tile + delta; // The tile beyond the bridge
+	RoadBits rcmd = DiagDirToRoadBits(ReverseDiagDir(bridge_dir)); // The road bits that would be built after extending the bridge.
+	RoadType rt = GetTownRoadType(t); // The road type being built by the town.
+
+	/* Before we try anything, make sure the tile is on the map and not the void. */
+	if (!IsValidTile(end_tile + delta)) return false;
```

Comment:
`next_tile = end_tile + delta;` so
```suggestion
	if (!IsValidTile(next_tile)) return false;
```

Reply:
Yes, thanks.